### PR TITLE
Pass token to requests in dataloader

### LINKED ISSUE #155: Use token-based auth for ERT Storage
Now that tokens are in, we should add token-based auth to webviz. Probably depends on https://github.com/equinor/ert/issues/1700

### PR DESCRIPTION
**Pull request by ManInFez**
resolves #155

### CODE REVIEW

File: webviz_ert/data_loader/__init__.py [hunk 1]
```diff
@@ -14,15 +14,17 @@
...
 
 
 # these are needed to mock for testing
-def _requests_get(*args: Union[str, bytes], **kwargs: Any) -> requests.models.Response:
+def _requests_get(*args: Any, **kwargs: Any) -> requests.models.Response:
     return requests.get(*args, **kwargs)
 
 
-def _requests_post(*args: Union[str, bytes], **kwargs: Any) -> requests.models.Response:
+def _requests_post(*args: Any, **kwargs: Any) -> requests.models.Response:
```
Comment (xjules) [inline webviz_ert/data_loader/__init__.py:27]: `Union[str, bytes,dict]`?

File: webviz_ert/data_loader/__init__.py [hunk 2]
```diff
@@ -150,6 +153,11 @@ def _query(self, query: str, **kwargs: Any) -> dict:
     def _get(
         self, url: str, headers: dict = None, params: dict = None
     ) -> requests.Response:
+        if headers is None:
+            headers = {"Token": self.token}
+        else:
+            headers.update({"Token": self.token})
+
         resp = _requests_get(f"{self.baseurl}/{url}", headers=headers, params=params)
```
Comment (dotfloat) [inline webviz_ert/data_loader/__init__.py:161]:
```suggestion
        if headers is None:
            headers = {}

        resp = _requests_get(f"{self.baseurl}/{url}", headers={**headers, "Token": self.token}, params=params)
```

1. Avoids code duplication
2. Avoids mutating input data

File: webviz_ert/data_loader/__init__.py (hunk 1 above)
Comment (ManInFez) [reply webviz_ert/data_loader/__init__.py:27]: fails with `error: Argument 1 to "post" has incompatible type "*Tuple[Union[str, bytes, Dict[Any, Any]], ...]"; expected "bool"  [arg-type]`

Comment (xjules) [reply webviz_ert/data_loader/__init__.py:27]: ok, then `Any` fits the bill.

**Review by dotfloat — APPROVED** [review]

**Pull request merged**